internal/builders/go/pkg/build_test.go

```diff
@@ -1047,3 +1048,114 @@ func Test_generateCommand(t *testing.T) {
 func asPointer(s string) *string {
 	return &s
 }
+
+func TestGoBuild_Run(t *testing.T) {
+	type fields struct {
+		cfg     *GoReleaserConfig
+		goc     string
+		argEnv  map[string]string
+		ldflags string
+	}
+	type args struct {
+		dry bool
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "valid flags",
+			fields: fields{
+				cfg: &GoReleaserConfig{
+					Goos:   "linux",
+					Goarch: "amd64",
+					Binary: "binary",
+					Main:   asPointer("../builders/go/main.go"),
+					Dir:    asPointer("../builders/go"),
+					Ldflags: []string{
+						"-X main.version=1.0.0",
+					},
+				},
+			},
+			args: args{
+				dry: true,
+			},
+		},
+		{
+			name: "valid flags",
+			fields: fields{
+				cfg: &GoReleaserConfig{
+					Goos:   "linux",
+					Goarch: "amd64",
+					Binary: "/tmp/binary",
+					Main:   asPointer("main.go"),
+					Dir:    asPointer("./testdata/go"),
+					Ldflags: []string{
+						"-X main.version=1.0.0",
+					},
+				},
+			},
+			args: args{
+				dry: false,
+			},
+		},
+		{
+			name: "slash in the binary name",
+			fields: fields{
+				cfg: &GoReleaserConfig{
+					Goos:   "linux",
+					Goarch: "amd64",
+					Binary: "tmp/binary",
+					Main:   asPointer("../builders/go/main.go"),
+					Dir:    asPointer("../builders/go"),
+				},
+			},
+			args: args{
+				dry: true,
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid flags",
+			fields: fields{
+				cfg: &GoReleaserConfig{
+					Goos:    "linux",
+					Goarch:  "amd64",
+					Binary:  "binary",
+					Main:    asPointer("../builders/go/main.go"),
+					Dir:     asPointer("../builders/go"),
+					Ldflags: []string{},
+				},
+			},
+			args: args{
+				dry: true,
```

Comment:
Do you want to maybe add non-dry versions of these tests? or do they need to be dry for other reasons?